language/compiler/ir-to-bytecode/syntax/src/ast.rs

```diff
@@ -291,6 +292,8 @@ pub struct Function {
     /// This list of acquires grants the borrow checker the ability to statically verify the safety
     /// of references into global storage
     pub acquires: Vec<StructName>,
+    /// List of postconditions
```

Comment:
What about preconditions?

Reply:
I would just make this a list of Condition, and then have the condition decide what it is (succeeds-if, aborts-if, ensures, etc.)

Reply:
Ah, I now understand your previous comment :). Will do it that way.